graphql_client_codegen/src/operations.rs

```diff
@@ -113,3 +113,41 @@ impl ::std::convert::From<OperationDefinition> for Operation {
         }
     }
 }
+
+impl<'a> ::std::convert::From<&'a OperationDefinition> for Operation {
+    fn from(definition: &OperationDefinition) -> Operation {
+        match *definition {
+            OperationDefinition::Query(ref q) => Operation {
+                name: q.name.clone().expect("unnamed operation"),
+                operation_type: OperationType::Query,
+                variables: q
+                    .variable_definitions
+                    .iter()
+                    .map(|v| v.clone().into())
+                    .collect(),
+                selection: (&q.selection_set).into(),
+            },
+            OperationDefinition::Mutation(ref m) => Operation {
+                name: m.name.clone().expect("unnamed operation"),
+                operation_type: OperationType::Mutation,
+                variables: m
+                    .variable_definitions
+                    .iter()
+                    .map(|v| v.clone().into())
+                    .collect(),
+                selection: (&m.selection_set).into(),
+            },
+            OperationDefinition::Subscription(ref s) => Operation {
+                name: s.name.clone().expect("unnamed operation"),
+                operation_type: OperationType::Subscription,
+                variables: s
+                    .variable_definitions
+                    .iter()
+                    .map(|v| v.clone().into())
+                    .collect(),
+                selection: (&s.selection_set).into(),
+            },
+            OperationDefinition::SelectionSet(_) => panic!(SELECTION_SET_AT_ROOT),
+        }
+    }
+}
```

Comment:
this is mostly copy-pasted, ideally I'll find time to clean it up before merging